go/src/koding/kites/kloud/machine/mongo.go

```diff
@@ -105,12 +107,19 @@ func (m *MongoDatabase) Machines(f *Filter) ([]*Machine, error) {
 		groupTitles[st.Id] = [2]string{st.Group, st.Title}
 	}
 
+	var team, stack string
 	machines := make([]*Machine, len(machinesDB))
 	for i, mdb := range machinesDB {
+		team, stack = "", ""
+		if mdb.GeneratedFrom != nil {
+			team = groupTitles[mdb.GeneratedFrom.TemplateId][0]
+			stack = groupTitles[mdb.GeneratedFrom.TemplateId][1]
+		}
+
 		machines[i] = &Machine{
 			ID:          mdb.ObjectId.Hex(),
-			Team:        groupTitles[mdb.GeneratedFrom.TemplateId][0],
-			Stack:       groupTitles[mdb.GeneratedFrom.TemplateId][1],
+			Team:        team,
+			Stack:       stack,
```

Comment:
Nit: You could assign to machine[i].Stack directly without using temporary vars.